src/components/layouts/php-collection-page-layout.tsx

```diff
@@ -0,0 +1,713 @@
+import LearnerRatings from '../pages/courses/learner-ratings'
+import CommunityResource from 'components/community-resource'
+import TagList from './tag-list'
+import DialogButton from '../pages/courses/dialog-button'
+import MembershipDialogButton from '../pages/courses/membership-dialog-button'
+
+import {
+  logCollectionResource,
+  Duration,
+  PeopleCompleted,
+  StarsRating,
+  PublishedAt,
+  UpdatedAt,
+} from './collection-page-layout'
+
+import LoginForm from 'pages/login'
+
+type CoursePageLayoutProps = {
+  lessons: any
+  course: any
+  ogImageUrl: string
+}
+
+const LessonLinkResource = ({
+  lesson,
+  completedLessonSlugs,
+  index,
+}: {
+  lesson: any
+  completedLessonSlugs: any
+  index: number
+}) => {
+  const isComplete = completedLessonSlugs.includes(lesson.slug)
+  return (
+    <li key={lesson.slug}>
+      <div className="flex py-2 font-semibold leading-tight h-20">
+        <div className="flex items-center mr-2 space-x-2">
+          <div
+            className={`${
+              isComplete
+                ? 'text-blue-600 dark:text-green-400'
+                : 'text-gray-500 dark:text-gray-400'
+            } pt-px font-xs scale-75 font-normal w-4`}
+          >
+            {isComplete ? (
+              <CheckIcon className="w-6 h-6 -translate-x-2" />
+            ) : (
+              index + 1
+            )}
+          </div>
+          {lesson.icon_url && (
+            <div className="flex items-center flex-shrink-0 w-8">
+              <Image src={lesson.icon_url} width={24} height={24} />
+            </div>
+          )}
+        </div>
+        {lesson.path && (
+          <div className="flex flex-col ">
+            <div>
+              <Link href={lesson.path}>
+                <a className="text-lg font-semibold hover:underline hover:text-blue-600 dark:text-gray-100">
+                  {lesson.title}
+                </a>
+              </Link>
+            </div>
+            <div className="text-xs text-gray-700 dark:text-gray-500">
+              {convertTimeWithTitles(lesson.duration, {
+                showSeconds: true,
+              })}
+            </div>
+          </div>
+        )}
+      </div>
+    </li>
+  )
+}
+
+const ModuleCollection = ({module, children}: {module: any; children: any}) => {
+  return (
+    <div key={module.id}>
+      <h2 className="text-xl font-bold mt-4 mb-2 sm:mx-0 mx-auto w-fit">
+        {module.title}
+      </h2>
+      <div>
+        <p className="mb-6 prose text-gray-900 dark:prose-dark md:prose-lg md:dark:prose-lg-dark dark:text-gray-100 dark:prose-a:text-blue-300 dark:hover:prose-a:text-blue-200 prose-a:text-blue-500 hover:prose-a-:text-blue-600 ">
+          {module.description}
+        </p>
+        <ul
+          className="grid sm:grid-flow-col gap-4"
+          style={{
+            gridTemplateRows: `repeat(${Math.ceil(
+              module.resourceList.length / 2,
+            )}, minmax(0, 1fr))`,
+          }}
```

Comment:
The only solution I found is to use inline styles here.
Arbitrary values for custom tailwindcss classnames should be predefined and explicitly written, so we can't use something like 
```
grid-rows-[repeat(${Math.ceil(module.resourceList.length / 2)},_minmax(0,_1fr))]
```